### PR TITLE
Fix to issue 31502: problem with FakeHLT in PU 2017-2018 relvals

### PR DESCRIPTION
**Pull request by jfernan2**
#### PR description:

Upon creation of PR https://github.com/cms-sw/cmssw/pull/31472 202X upgrade samples were wrongly assigned FakeHLT Menu configurations for DQM/Validation steps.

This step fixes the issue as reported on https://github.com/cms-sw/cmssw/issues/31502

#### PR validation:

Validated with 
runTheMatrix.py --list=11834.99,12634.99,13034.99

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No

### CODE REVIEW

**Comment by jfernan2** [comment]
solves https://github.com/cms-sw/cmssw/issues/31502

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test workflow 11834.99,12634.99,13034.99

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -927,6 +927,10 @@ def condition(self, fragment, stepList, key, hasHarvest):
     upgradeProperties[2017][key+'PU'] = deepcopy(upgradeProperties[2017][key])
     upgradeProperties[2017][key+'PU']['ScenToRun'] = ['GenSim','DigiPU','RecoFakeHLTPU','HARVESTFakeHLTPU'] + \
                                                      (['Nano'] if 'Design' not in key else [])
+    if '202' in key:   #2021, 2023 and 2023 still run not FakeHLT Menus
+               upgradeProperties[2017][key+'PU']['ScenToRun'] = ['GenSim','DigiPU','RecoPU','HARVESTPU'] + \
+                                                     (['Nano'] if 'Design' not in key else [])
+
```
Comment (kpedro88) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:933]:
```suggestion
    upgradeProperties[2017][key+'PU']['ScenToRun'] = ['GenSim','DigiPU'] + \
                                                     (['RecoPU','HARVESTPU'] if '202' in key else ['RecoFakeHLTPU','HARVESTFakeHLTPU']) + \ #2021, 2023 and 2023 still run not FakeHLT Menus
                                                     (['Nano'] if 'Design' not in key else [])
```

**Comment by jfernan2** [comment]
please test workflow 11834.99,12634.99,13034.99

**Comment by jfernan2** [comment]
please test workflow 11834.99,12634.99,13034.99

**Comment by qliphy** [comment]
type bug-fix

**Comment by mmusich** [comment]
hello @kpedro88 @chayanit, @wajidalikhan, @kpedro88, @jordan-martins
can you please sign and integrate this PR at earliest convenience? It's stopping some developments for lack of a working premixed workflow for testing.

**Comment by boudoul** [comment]
a more explicit title would be appreciated, thank you .

**Comment by silviodonato** [comment]
urgent

**Comment by chayanit** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**